src/backend/distributed/commands/alter_table.c

```diff
@@ -1183,6 +1196,72 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 }
 
 
+/*
+ * AnyColumnsRequireDataCopy decides if relation has any columns that we
+ * might need to copy the data of when replacing table.
+ */
+static bool
+AnyColumnsRequireDataCopy(Oid relationId)
+{
+	return list_length(GetNonGeneratedStoredColumnNameList(relationId)) > 0;
+}
+
+
+/*
+ * BuildInsertColumnStringForReplaceTable is a helper function that returns
+ * comma seperated columns that we can insert into for ReplaceTable.
+ */
+static char *
+BuildInsertColumnStringForReplaceTable(Oid relationId)
+{
+	/*
+	 * When copying data to temp table, we skip columns having GENERATED
+	 * GENERATED ALWAYS AS (...) STORED expressions since Postgres doesn't
+	 * allow inserting into such columns. This is not bad since Postgres
+	 * would already generate such columns.
+	 */
```

Comment:
might be useful to have a comment on why we don't filter out default columns?

Reply:
dc76a40785f9bfc4e48255177fcae0da6c332f41